x-pack/plugins/infra/public/pages/metrics/hosts/components/tabs/alerts_tab_badge.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { EuiToolTip } from '@elastic/eui';
+import { EuiNotificationBadge } from '@elastic/eui';
+import { EuiIcon } from '@elastic/eui';
+import { EuiLoadingSpinner } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { useAlertsCount } from '../../../../../hooks/use_alerts_count';
+import { infraAlertFeatureIds } from './config';
+import { useAlertsQuery } from '../../hooks/use_alerts_query';
+
+export const AlertsTabBadge = () => {
+  const { alertsEsQuery } = useAlertsQuery();
+
+  const { alertsCount, loading, error } = useAlertsCount({
+    featureIds: infraAlertFeatureIds,
+    filter: alertsEsQuery,
+  });
+
+  if (loading) {
+    return <EuiLoadingSpinner />;
+  }
+
+  if (error) {
+    return (
+      <EuiToolTip
+        content={i18n.translate('xpack.infra.hostsViewPage.tabs.alerts.countError', {
+          defaultMessage:
+            'The active alert count was not retrieved correctly, try reloading the page.',
+        })}
+      >
+        <EuiIcon color="warning" type="alert" />
+      </EuiToolTip>
+    );
+  }
+
+  return (
+    <EuiNotificationBadge className="eui-alignCenter" size="m">
```

Comment:
Looks like the count gets messed up if I flip between the status filters.


https://user-images.githubusercontent.com/4104278/217802943-13374b70-5058-4ac9-8fa1-9607be730ec9.mp4

Reply:
I noticed the issue regarding the de-sync between the count going to 0 when we instead have active alerts and I pushed already a fix, try to pull the changes and let me know if now it works fine :)

Regarding the count when switching to the **Recovered** filter, the count always represents the count of active alerts, not the recovered ones, that's why it keeps rendering the count of the actives. 
This means that, in case the user switch back to the metrics tab, she/he will still be able to see the active alerts count.

Reply:
Tested again and it seems to be fixed on my end 👍

Reply:
> Regarding the count when switching to the Recovered filter, the count always represents the count of active alerts, not the recovered ones, that's why it keeps rendering the count of the actives.

Sorry, didn't mean to imply that this wasn't correct in my screen recording 🙂